### PR TITLE
Kotlin decode from byte array

### PR DESCRIPTION
**Pull request by Fejiro001**
### Description

- Added the .decodeFromByteArray() method for Kotlin Serialization
- Closes Issue #3869 

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by dakshdeepHERE** [comment]
@Fejiro001 remove the identical file from the datetime folder please. We need it only in the serialisation folder

File: content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  - 'Method'
+  - 'Data Structures'
+  - 'Time'
+  - 'Date'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+The **`.decodeFromByteArray()`** method is a part of the `kotlinx.serialization` library and is used to deserialize/decode (convert) byte array into a Kotlin object. Commonly used when data in byte array format is transmitted and needs to be reverted back into a Kotlin object.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md:17]:
```suggestion
The **`.decodeFromByteArray()`** method is a part of the **kotlinx.serialization** library and is used to deserialize/decode (convert) byte array into a Kotlin object. Commonly used when data in byte array format is transmitted and needs to be reverted back into a Kotlin object. 
```

File: content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+## Syntax
+
+```pseudo
+fun <T> BinaryFormat.decodeFromByteArray(ByteArray)
+```
+
+- `<T>`: the type of the object to be deserialized
+- `BinaryFormat`: an interface in `kotlinx.serialization` library that provides methods for serializing and deserializing Kotlin objects in binary formats
+- `ByteArray`: the byte array containing the binary representation of the serialized object
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md:27]:
```suggestion
- `<T>`: The type of the object to be deserialized
- `BinaryFormat`: An interface in `kotlinx.serialization` library that provides methods for serializing and deserializing Kotlin objects in binary formats
- `ByteArray`: The byte array containing the binary representation of the serialized object
```

File: content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+## Example
+
+The following example demonstrates how to use `.decodeFromByteArray`.
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md:31]:
```suggestion
The following example demonstrates how to use `.decodeFromByteArray`:
```

File: content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+@Serializable
+data class Person(val name: String, val age: Int)
+
+fun main() {
+    val person = Person("Alice", 25)
+    val personToByteArray = ProtoBuf.encodeToByteArray(person)
+    val deserializedPerson = ProtoBuf.decodeFromByteArray(personToByteArray)
+    println("Original Person: $person")
+    println("Deserialized Person: $deserializedPerson")
+}
```
Comment (dakshdeepHERE) [inline content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md:45]:
```suggestion
import kotlinx.serialization.*

@Serializable
data class Person(val name: String, val age: Int)

fun main() {
  val person = Person("Alice", 25)
  val personToByteArray = ProtoBuf.encodeToByteArray(person)
  val deserializedPerson = ProtoBuf.decodeFromByteArray(personToByteArray)
  println("Original Person: $person")
  println("Deserialized Person: $deserializedPerson")
}
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Some initial comments for you to work on @Fejiro001. 🚀

**Review by dakshdeepHERE — APPROVED** [review]
This looks good for second review @Fejiro001 🚀

File: content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+```kotlin
+import kotlinx.serialization.*
+
+@Serializable
+data class Person(val name: String, val age: Int)
+
+fun main() {
+  val person = Person("Alice", 25)
+  val personToByteArray = ProtoBuf.encodeToByteArray(person)
```
Comment (SSwiniarski) [inline content/kotlin/concepts/serialization/terms/decodeFromByteArray/decodeFromByteArray.md:41]: What are these ProtoBuf references? They aren't declared anywhere.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Fejiro001 I left a few comments. Also, please make sure that your code is runnable as written. There are currently unresolved references in it.

**Review by SSwiniarski — APPROVED** [review]
@Fejiro001 looks good.

**Pull request merged**